site-src/geps/gep-992.md

```diff
@@ -0,0 +1,184 @@
+	// filters specified that would result in a response being sent, a HTTP 503
+	// status code is returned.
+	//
+	// Support: Core for Kubernetes Service
+	// Support: Custom for any other resource
+	//
+	// Support for weight: Core
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=16
+	BackendRefs []HTTPBackendRef `json:"backendRefs,omitempty"`
+
+	// LoadBalancerStrategy defines the high level strategy to use when multiple
+	// BackendRefs are configured for traffic.
+	//
+	// If only one backend is provided in BackendRefs this property has no
+	// effect, otherwise it is required.
+	//
+	// Support: Core for RoundRobinStrategy
+	// Support: Core for WeightedStrategy
+	// Support: Extended for LeastConnectionsStrategy
+	//
+	// +optional
+	// +kubebuilder:validation:Enum=RoundRobin,WeightedStrategy,LeastConnectionsStrategy
+	// +kubebuilder:default=RoundRobin
+	LoadBalancerStrategy *LoadBalancerStrategy `json:"loadBalancerStrategy,omitempty"`
+}
+```
+
+Practically speaking this changes very little about how things already work,
+except that end-users and implementers can now very clearly understand the
+intended dispersement of traffic across backends instead of the implicit
+nature of the previous model.
+
+We document the individual strategies and more can be added over time:
+
+```golang
+const (
+	// RoundRobin is a simple LoadBalancerStrategy which distributes inbound
+	// matching requests across all HTTPBackendRefs evenly according to the
+	// overall number of requests.
+	RoundRobinStrategy LoadBalancerStrategy = "RoundRobin"
+
+	// WeightedStrategy distributes inbound matching requests across
+	// HTTPBackendRefs according to the "weight" value set on each individual
+	// ref, and uses those weights to determine what percentage of traffic each
+	// backend should take on.
+	WeightedStrategy LoadBalancerStrategy = "Weighted"
+
+	// LeastConnectionsStrategy distributes inbound matching requests across all
+	// HTTPBackendRefs in such a way as to try and maintain an even number of open
+	// connections across all refs.
+	LeastConnectionsStrategy LoadBalancerStrategy = "LeastConnections"
+)
+```
+
+Additionally the `weight` on individual `HTTPBackendRef` objects becomes
+required when `WeightedStrategy` is in use, and invalid when using other
+strategies.
+
+With these changes, it's now clear and present how traffic will be dispersed
+and we can continue to add more options as the need arises:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: multiple-backends
+spec:
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /bar
+    backendRefs:
+    - name: service1
+      port: 80
+    - name: service2
+      port: 80
+    - name: service3
+      port: 80
+    loadBalancerStrategy: "LeastConnections"
+```
+
+TODO: determine the best way to add "NamedStrategy" which would allow
+      individual implementations to define their own strategies.
```

Comment:
Agree with this. If we're going to take the route of built in configuration, we'll need to leave room for custom types + likely custom configuration to go along with that.